cpp/src/c_api/random_walks.cpp

```diff
@@ -156,7 +157,7 @@ struct uniform_random_walks_functor : public cugraph::c_api::abstract_functor {
   cugraph::c_api::cugraph_graph_t* graph_{nullptr};
   cugraph::c_api::cugraph_type_erased_device_array_view_t const* start_vertices_{nullptr};
   size_t max_length_{0};
-  size_t seed_{0};
+  cugraph::c_api::cugraph_rng_state_t* rng_state_{nullptr};
```

Comment:
Can you add a FIXME comment here to indicate that the rng state should be passed in as a parameter.  We can make that a separate PR, since it will affect a bunch more code (including python and PLC calls).